app/lib/hyku_addons/per_tenant_smtp_interceptor.rb

```diff
@@ -8,9 +8,9 @@ def self.available_smtp_fields
 
     def self.delivering_email(message)
       Account.find_by(tenant: Apartment::Tenant.current)&.switch!
-      return unless (mailer_settings = Settings.smtp_settings.presence)
+      return unless (mailer_settings = Settings.smtp_settings).present?
```

Comment:
Its best not to negate a positive, so this should be `return if ().blank?`

Reply:
We are trying to both do the conditional clause evaluation and assign it to a variable in a single step.

You can't use the `present?` method for that as it returns a truthy or falsey instead of the hash we are interested on, we need to use that hash to do some stuff later.

Reply:
@okeen `.presence` would return the value or `nil`, so the only reason i can see that it would be needed inside the `()` is that you don't know if `smtp_settings` isn't actually set or you need to add an additional stage to the condition `...presence || backup_var`. 

>You can't use the present? method for that as it returns a truthy or falsey instead of the hash we are interested on, we need to use that hash to do some stuff later.

If you're just assigning the value and want to check its not `nil`, then using `.present?` is clearer when reading the code IMO - I use it a lot for single line guard statements that also assign a variable. 

Its the brackets that that work out the order of evaluation... `foo = bar.present?` would be `true`/`false`, but `(foo = bar).present?` assigns `foo` to the value of `bar` and checks if they are present.